src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs

```diff
@@ -75,20 +77,31 @@ public static void Initialize()
                 string loaderFilePath = Path.Combine(loaderAssemblyLocation, "OpenTelemetry.AutoInstrumentation.Loader.dll");
                 Assembly loaderAssembly = Assembly.LoadFrom(loaderFilePath);
                 loaderAssembly.CreateInstance("OpenTelemetry.AutoInstrumentation.Loader.Startup");
-                StartupHookEventSource.Log.Trace("StartupHook initialized successfully!");
+                Logger.LogTrace("StartupHook initialized successfully!");
             }
             else
             {
-                StartupHookEventSource.Log.Trace("OpenTelemetry.AutoInstrumentation.Instrumentation initialized before startup hook");
+                Logger.LogTrace("OpenTelemetry.AutoInstrumentation.Instrumentation initialized before startup hook");
             }
         }
         catch (Exception ex)
         {
-            StartupHookEventSource.Log.Error($"Error in StartupHook initialization: LoaderFolderLocation: {loaderAssemblyLocation}, Error: {ex}");
+            Logger.LogError($"Error in StartupHook initialization: LoaderFolderLocation: {loaderAssemblyLocation}, Error: {ex}");
             throw;
         }
     }
 
+    private static void InitLogger()
+    {
+        // Workaround to make startup hook load the logger assembly.
+        var executionPath = Assembly.GetExecutingAssembly().Location;
+        var dirname = Path.GetDirectoryName(executionPath);
+        if (!string.IsNullOrEmpty(dirname))
+        {
+            Assembly.LoadFrom(Path.Combine(dirname, "OpenTelemetry.AutoInstrumentation.Logging.dll"));
```

Comment:
Shouldn't we catch exception thrown by `Assembly.LoadFrom` and use event source to have any kind of indcation that something bad happened?